### DISCUSSION TITLE
Buderus GB172i.2 (BC400) `flowtemp` support

### DISCUSSION THREAD

**Discussion by molszowy** [emsesp/EMS-ESP32#1755] [Q&A] [answered]
Hi all! I have a newly installed Buderus GB172i.2 with BC400 panel. I'm going to integrate it with Loxone, but would like to confirm whether this is going to work via the EMS gateway before I buy the hardware. 

Has anyone tested whether the `flowtemp` input parameter takes effect on such boiler with BC400 ? I understand it's normally set by the thermostat, however I don't have any from Buderus. The Loxone software offers ability to calculate the flow temperature based temperature settings from all rooms (sensors) so I would like to leverage that. I don't currently have a mixer so will control only the boiler itself.

Also, would setting 0 value, disable the boiler heating ?

Any help is highly appreciated.

**Comment by proddy** [comment]
I believe so. EMS-ESP can set the flowtemp via the heatsource module directly. Maybe other active users with a BC400 like @Bingo2023, @h3llsp4wn88 and @theFeverDog would know? I think your heat pump comes with an integrated remote RC thermostat which EMS-ESP detects too.

**Comment by Bingo2023** [comment]
I'm not sure where to find "flowtemp"...
But you will be able to control your GB172i.2 with EMS-ESP.
I'm using Nodered to control it. Home Assistant is only used to visualize it and change values manual, all intelligence is in Nodered.
I have an RC220 Thermostat next to the boiler installed, don't know if really needed. There are some activities to make the thermostat "virtual", guess that that feature come in future.
I'm getting the room temperatures via KNX, plan to use hottest & coldest room temperature and feed this into a virtual thermostat (for heating and cooling), but that is just a plan for the next month... But that is just a fun project as for new houses all this is not needed, the house is well isolated so hand control to turn on/off few times per year will also do it :)

here is just a example of Nodered to catch your attention:

![2024-05-09_09h13_38](https://github.com/emsesp/EMS-ESP32/assets/61052575/4af64523-60d5-491c-a36a-8217627bb009)


so: will it work out of the box? No ! But if you contribute and invest time, you will be able to make it work.

**Reply by proddy** [reply]
I'm seeing more NodeRed usage, something we should add to the Wiki. I love this posting on our Discord channel:

<img width="1090" alt="image" src="https://github.com/emsesp/EMS-ESP32/assets/1230712/6ca5c202-a9c1-438f-bbb3-acc58a2855a2">

**Comment by MichaelDvP** [comment, marked as answer]
@molszowy 
Yes, without  thermostat you can set the flowtemp with emsesp, many users do this. 
- the flowtemp on the controlpanel have to be set higher, the temperature send over the bus is only accepted if lower.
- you have to send at least every minute, otherwise the boiler fall back ti the higher setpoint from the control.
- we have a special entity `forceheatingoff` that sends every minute a `flowtemp 0` to the boiler, some users like to use a on/off control.

**Reply by proddy** [reply]
Thanks Michael, I've updated the wiki with your input

https://emsesp.github.io/docs/tips-and-tricks/#control-the-boiler-heating

**Reply by molszowy** [reply]
Thanks, this is helpful. 

> the flowtemp on the controlpanel have to be set higher, the temperature send over the bus is only accepted if lower.

I guess you mean the maximum flow temperature setting (currently 40 degrees for floor heating). On the main screen, I can see some target temperature in auto or manual mode (set to 21, cannot go higher than 30 degrees). This makes me think this is no the same as flow temperature. This is probably the one used to control the heating curve, something like desired room temperature. I don't see the way to set the flow temperature through control panel currently.

Is my understanding correct ? What's the parameter's name ?

**Reply by MichaelDvP** [reply]
The entity to control is `selflowtemp`, maximum flow temp is also a limit, but the limit and fallback is the `heatingtemp`
